Extend filtering of backtrace frames a bit

cole-miller opened the pull request:
This adds a filename and line and column numbers to each frame of the backtrace that's sent with panic reports. The full source file path is not sent, both for brevity and because this path might be considered sensitive information.

I also tweaked the condition for dropping uninformative frames related to panic handling at the top of the trace to reflect that we see both `rust_begin_unwind` and `_rust_begin_unwind` in practice (I don't know why).

Release Notes:

- N/A

ConradIrwin commented:
@cole-miller I think the line/file information is stripped from release builds (I tried it in 345b983c8e and reverted it in fbc693d1d7).

Altering the prefix stripping would be nice though.

On macOS we use the system crash reporter in addition to the panic handler, and then ./script/symbolicate can re-add the file-name/line information. https://zed.dev/docs/development/debugging-crashes.

For Linux, I think we should try and do something like this: https://bun.sh/blog/bun-report-is-buns-new-crash-reporter. Initially just in the panic hook, but ideally one day in a signal handler too. (we could use Crashpad, etc.., but I'd rather not if we don't have to).

cole-miller commented:
@ConradIrwin thanks, should've figured! I saw that we set
`profile.release.debug = "limited"` in Cargo.toml but missed the post-build
stripping in the build scripts. Will merge this with just the filtering change.

cole-miller committed:
Expand filtering of backtrace frames a bit

Both rust_begin_unwind and _rust_begin_unwind appear in practice, not
sure why.
